packages/format-library/src/text-color/index.js

```diff
@@ -59,15 +89,10 @@ function TextColorEdit( {
 				className="format-library-text-color-button"
 				isActive={ isActive }
 				icon={
-					<>
-						<Icon icon={ textColorIcon } />
-						{ isActive && (
-							<span
-								className="format-library-text-color-button__indicator"
-								style={ colorIndicatorStyle }
-							/>
-						) }
-					</>
+					<Icon
+						icon={ textColorIcon }
+						style={ colorIndicatorStyle }
+					/>
```

Comment:
Having the icon reflect color choices is pretty neat, though it is possible to create a bad combination where the icon is no longer legible:


<img width="451" alt="CleanShot 2021-09-17 at 10 12 51@2x" src="https://user-images.githubusercontent.com/1270189/133827965-fb9b1670-408f-401c-9df6-ee9946ee0866.png">
<img width="447" alt="CleanShot 2021-09-17 at 10 13 12@2x" src="https://user-images.githubusercontent.com/1270189/133827976-19476b6e-5453-4a1e-b457-823ca0b681e5.png">

Reply:
Right, it would be the same as the text though, making it an accurate preview. As long as the button label is readable, I find that ok.